src/tuner/nccl_ofi_tuner.c

```diff
@@ -305,25 +340,30 @@ ncclResult_t nccl_ofi_tuner_get_coll_info(
 		goto exit;
 	}
 
+	float(*table)[NCCL_NUM_PROTOCOLS] = (float(*)[NCCL_NUM_PROTOCOLS])collCostTable;
 	int in_out = -1;
+	int algorithm = NCCL_ALGO_UNDEF;
+	int protocol = NCCL_PROTO_UNDEF;
 	nccl_ofi_tuner_point_t p = {.x = nBytes, .y = nccl_ofi_tuner_ctx->dims.num_ranks};
 
 	/* Check all regions */
 	for (int i = 0; i < nccl_ofi_tuner_ctx->num_regions && in_out < 0; i++) {
-		if (nccl_ofi_tuner_ctx->regions[i].algorithm == NCCL_ALGO_NVLS_TREE && nvlsSupport == 0) {
+		algorithm = nccl_ofi_tuner_ctx->regions[i].algorithm;
+		protocol = nccl_ofi_tuner_ctx->regions[i].protocol;
+		if (table[algorithm][protocol] == NCCL_ALGO_PROTO_IGNORE || algorithm >= numAlgo ||
+		    protocol >= numProto) {
 			continue;
 		}
 
 		in_out = is_inside_region(p, &nccl_ofi_tuner_ctx->regions[i]);
 		if (in_out >= 0) {
-			*algorithm = nccl_ofi_tuner_ctx->regions[i].algorithm;
-			*protocol = nccl_ofi_tuner_ctx->regions[i].protocol;
+			table[algorithm][protocol] = 0.0;
 
 			NCCL_OFI_INFO(NCCL_TUNING,
 				      "Choosing algo %d proto %d with cost %.8f µsecs for coll %d size %ld.",
```

Comment:
Yes, but I don't know for how long we will keep this approach or when we will go back to a model-based one. 
And I don't want to change log statements too often, since people could have regexes targeting them.